lib/msf/core/web_services/json_rpc_exception_handling.rb

```diff
@@ -0,0 +1,34 @@
+require 'msf/core/rpc'
+
+module Msf::WebServices
+  module JsonRpcExceptionHandling
+
+    def self.registered(app)
+      app.set :show_exceptions => false
+
+      app.error { |err|
+        self.request.body.rewind
+        request = self.request.body.read
+
+        begin
+          request = JSON.parse(request, symbolize_names: true)
```

Comment:
this is a bit confusing using the name `request` to refer to the request body, especially with self.request thrown into the mix aswell

Reply:
Agreed, changing it to `initial_request`